### ISSUE TITLE
Documentation clarification - bidir, OneToMany

### ISSUE THREAD

**Issue by bitwombat** [doctrine/doctrine2#6083]
From [the reference manual](http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/reference/association-mapping.html#one-to-many-bidirectional):

> A one-to-many association has to be bidirectional, unless you are using an additional join-table. This is necessary, because of the foreign key in a one-to-many association being defined on the “many” side. Doctrine needs a many-to-one association that defines the mapping of this foreign key.

I know this is basic and fundamental, and I appreciate you guys' patience.  But I don't follow the reasoning here.

> A one-to-many association has to be bidirectional

OK, that's the statement, and what I don't understand.  The rest of the paragraph is the reasoning.

> This is necessary because of the foreign key in a one-to-many association being defined on the “many” side.

I understand that the FK is defined on the many side.  But why does this make bidirectionality necessary?

> Doctrine needs a many-to-one association that defines the mapping of this foreign key.

But, all the information is present in the "many" side:

```
    // ...
    /**
     * @ManyToOne(targetEntity="Product", inversedBy="features")
     * @JoinColumn(name="product_id", referencedColumnName="id")
     */
    private $product;
```

Why do we need the mirror image of this same information on the "one" side?

Happy to help clarify the documentation with a PR if it's not "just me" :)

**Comment by coudenysj** [comment]
The mapping of the FK is defined on the many side and it should always be present on that side (see [A many-to-one association is the most common association between objects.](http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/reference/association-mapping.html#many-to-one-unidirectional).

If you want to use a one-to-many relation, you need to extend that many-to-one information and "replicate/inverse" the mapping on the one side. It is not possible to only define the mapping on the many side (because no information is stored on that side).

**Comment by bitwombat** [comment]
Thanks @coudenysj , that makes perfect sense.

While the information isn't _stored_ on the inverse (one) side, it could be reconstructed from the many side, right?

**Comment by coudenysj** [comment]
Correct.

**Comment by bitwombat** [comment]
OK, could be reconstructed, so the information is available, it might just be a pain for Doctrine to do.  So I'm not sure the reasoning in the documentation is correct.  Is "Define it on the many side so the code doesn't have to go through hoops figuring things out" more accurate?

**Comment by lcobucci** [comment]
@bitwombat you're right that it can be improved specially when we have the [5.6. One-To-Many, Unidirectional with Join Table](http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/reference/association-mapping.html#one-to-many-unidirectional-with-join-table) after it.

Would you like to send PR with your suggestions so we can debate there and already improve it?

**Comment by bitwombat** [comment]
@lcobucci Will do.  I know very little about the doctrine innards, so it will be from the viewpoint of an unsophisticated user :)

**Comment by lcobucci** [comment]
@bitwombat That's perfect since you're not biased by the knowledge 😜 thanks!

**Comment by bitwombat** [comment]
I'm re-reading this, hoping to do a PR to possibly clarify the docs, and I'm just not getting it still.

If I think of the unidirectional ManyToOne scenario, my "Many" entity has a scalar property pointing to the FK of the "One" entity.  Done.

With the unidirectional OneToMany scenario, my "One" entity has an array filled with FKs pointing to the "Many" entities.  But in a relational database, we don't have arrays... we have join tables.

So there's a join table mapping id's of my One to id's of my Many.

Doesn't that all work without making it bidir (adding a property with an InversedBy attribute on the Many side)?

**Comment by bitwombat** [comment]
Greetings to the two guys I managed to flypaper into this issue.  I'm still keen to understand this and (if still necessary) clarify the docs.  Perhaps my last comment took the air out of it.  If so, can you please re-read the thread and try to straighten me out?

**Comment by jelomada** [comment]
I'm not sure I can help here but you asked, so I will try to explain how I see it.

> With the unidirectional OneToMany scenario, my "One" entity has an array filled with FKs pointing to the "Many" entities. But in a relational database, we don't have arrays... we have join tables.

I'm not certain this is right way of thinking. 
If you will ask me about one-to-many or many-to-one relation in sql database (pure sql structure, not ORM) I will see two tables with foreign key in one of them. I won't see third join table. 
So I think one-to-many association shouldn't use join table by default. If that's what you suggest?
ManyToOne and OneToMany for me is the same thing but seeing from different sides. If you look at it from "many table" you have ManyToOne, if you look at it from "one table" you have OneToMany.

Back to Doctrine. 
I understand this like that:
Doctrine can not handle OneToMany unidirectional association. Period. It must have annotation on "many" side, so if you want to have access to "many" objects from "one" object you have to add annotation on "many" side. And this makes association bidirectional. And that is what documentation trying to deliver, but 5.5 and 5.6 may be a little bit confusing.

Perhaps a good idea is to add header (before 5.5) "One-To-Many, Unidirectional" and there explain it clearly:
"There is now way to manage OneToMany Unidirectional association by Doctrine. This is because the many side in a one-to-many association holds the foreign key, making it the owning side. Doctrine needs the many side defined in order to understand the association.
If you need access to "many" side from "one" side of your OneToMany association, you have two options: 
1) make association Bidirectional, that means add declaration on "many" side, look here: [ManyToOne Bidirectional](http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/reference/association-mapping.html#one-to-many-bidirectional)
2) use extra join table for holding relationship, this mean you will use [ManyToMany Unidirectional](http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/reference/association-mapping.html#many-to-many-unidirectional) association."

And then I would remove "5.6. One-To-Many, Unidirectional with Join Table" section because this is simply the same as ManyToMany Unidirectional.

**Comment by bitwombat** [comment]
Fixed in #6456
